config/environments/production.rb

```diff
@@ -90,9 +90,16 @@
   config.action_mailer.perform_caching = false
 
   # E-mails
+  outgoing_mail_domain = begin
+    Mail::Address.new(ENV['SMTP_FROM_ADDRESS']).domain
```

Comment:
Move `ENV.fetch('SMTP_FROM_ADDRESS', 'notifications@localhost')` to avoid having to use a fallback in Message-ID

No production server is supposed to run without SMTP_FROM_ADDRESS so fallback to web_domain is not needed

(Or, use web_domain as fallback for SMTP_FROM_ADDRESS in the first place)